slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -97,9 +97,10 @@ trait SQLiteProfile extends JdbcProfile {
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
       /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z]+)(\\(([0-9]+)\\))?$".r
+      final val TypePattern = """^([A-Z]+)(|\(([0-9]+)(|,([0-9]+))\))?$""".r
```

Comment:
Adding `|` does not make it optional, but allows, it to be empty, adding first `|` and giving input `VARCHAR` without braces there will still be capture group 2, just empty, removing first one does not have any impact as 2nd capture group is ignored, so it does not matter if it is empty string or null (that it was before).